chore: @metamask/swaps-controller v9 -> v10

nikoferro opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR includes the necessary changes for the client to work with @metamask/swaps-controller v10.0.0 (https://www.npmjs.com/package/@metamask/swaps-controller/v/10.0.0)

Related PRs
Upgrade controller to BaseController V2 ([#277](https://github.com/MetaMask/swaps-controller/pull/277))
Update name of Polygon network token from MATIC to POL ([#312](https://github.com/MetaMask/swaps-controller/pull/312))

These changes are needed since swaps-controller v10, will be upgrading to extend from base controller v2, changing the way how controllers interact with each other


## **Related issues**

N/A

## **Manual testing steps**

N/A

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

nikoferro committed:
chore: updates needed to support swaps-controller that uses base-controller v2

nikoferro committed:
chore: bump swaps-controller

nikoferro committed:
feat: basecontrollerv2 related changes

infiniteflower commented:
Need to delete the patch introduced in https://github.com/MetaMask/metamask-mobile/pull/10960

SwapsController v10 will have this: https://github.com/MetaMask/swaps-controller/pull/312

nikoferro committed:
chore: real yarn setup on v10

nikoferro committed:
temp ts fix

nikoferro committed:
file import fix

nikoferro committed:
lint fix

nikoferro committed:
removed patch

nikoferro commented:
> Need to delete the patch introduced in #10960
> 
> SwapsController v10 will have this: [MetaMask/swaps-controller#312](https://github.com/MetaMask/swaps-controller/pull/312)

Just updated the PR and removed this.

martahj reviewed (dismissed):
Have not QA'd it but code lgtm!

nikoferro committed:
fix for missing allowed calls

nikoferro committed:
fix for asset overview

legobeat commented on app/core/Engine.ts:
```diff
@@ -1554,33 +1555,43 @@ class Engine {
+          allowedEvents: [],
+        }),
+        // TODO: Remove once GasFeeController exports this action type
+        fetchGasFeeEstimates: () => gasFeeController.fetchGasFeeEstimates(),
```
Is this already done?


(`@metamask/gas-fee-controller@15.1.2`)

https://github.com/MetaMask/core/blob/d30f7df58f403cc9c68e4fe0652abf7320dbd852/packages/gas-fee-controller/src/GasFeeController.ts#L230

Or is this refering to something else?

nikoferro replied:
Its quite similar... the one you are proposing is the action to get the state, but not the one that triggers fetching the estimates on GasFeeController. 

I do see that the GasFeeController has a polling mechanism, but would need to dig deeper if getting the latest state of the controller is enough without manually triggering the fetch first (worth mentioning that the fetch function does return the new values, so we are not consuming this from the GasFeeController state directly)

If this is the case, i would keep it as a quick TODO for the next iteration, since changing this now, involves a new SwapsController release also. So i would prefer to merge v10.0.0 first, and then we can update (if this is the case), the controller with reading directly from the GasFeeController state from within the SwapsController on the next major

Daniel-Cross reviewed (changes requested):
This looks good but there's little to no checks or tests ran against these new changes.

At the very least, please run `E2E smoke tests` on the PR. I already added the label, you may need to remove and add again to get it to run.

I would also like to see some test coverage on the changes. At the moment I just have to trust this doesn't break anything. 

Also, a few conflicts that need addressing.

infiniteflower committed:
chore: reset yarn lock

infiniteflower committed:
chore: update yarn lock

infiniteflower committed:
fix: lint error

infiniteflower commented:
Tested

Daniel-Cross reviewed (approved):
Looks good!

jclancy93 reviewed (approved):
Tested locally and everything LGTM!
